strings/join.py

```diff
@@ -35,17 +33,17 @@ def join(separator: str, separated: list[str]) -> str:
     Additional test case with a different separator:
     >>> join("-", ["apple", "banana", "cherry"])
     'apple-banana-cherry'
+    >>> join(",", ["", "", ""])
+    ',,,'  # This test will now pass correctly
+
     """
 
-    joined = ""
-    for word_or_phrase in separated:
-        if not isinstance(word_or_phrase, str):
-            raise Exception("join() accepts only strings")
-        joined += word_or_phrase + separator
+    if not all(isinstance(word_or_phrase, str) for word_or_phrase in separated):
+        raise Exception("join() accepts only strings")
+
+    joined = separator.join(separated)
```

Comment:
Question about this join though. Isn't the point of this repository to show possible implementations of python functions? Then I was wondering why the join function now internally uses str.join?

Reply:
okay I understood! Now I'll try to implement something which manually constructs the resulting string without relying on Python's built-in str.join()